apps/aecontract/src/aect_channel_contract.erl

```diff
@@ -76,7 +76,12 @@ run_new(ContractPubKey, Call, CallData, Trees0, OnChainTrees,
 prepare_init_call(VmVersion, Protocol, Contract, Trees0) when ?IS_FATE_SOPHIA(VmVersion) ->
     Code = #{ byte_code := ByteCode } = aeser_contract_code:deserialize(aect_contracts:code(Contract)),
     FateCode  = aeb_fate_code:deserialize(ByteCode),
-    FateCode1 = aeb_fate_code:strip_init_function(FateCode),
+    FateCode1 =
+        if
+        % Before FATE 2 we do not include init function in code put in trees
+            VmVersion == ?VM_FATE_SOPHIA_1 -> aeb_fate_code:strip_init_function(FateCode);
```

Comment:
Move the check one line higher so for FATE2 we avoid unnecessary de-serialization and serialization of the code.